tests/kernel/pthread/src/pthread.c

```diff
@@ -173,14 +173,23 @@ int barrier_test_done(void)
 
 void main(void)
 {
-	int i, ret, status = TC_FAIL;
+	int i, ret, min_prio, max_prio, status = TC_FAIL;
 	pthread_attr_t attr[N_THR];
 	struct sched_param schedparam;
 	pthread_t newthread[N_THR];
 	int schedpolicy = SCHED_FIFO;
 	void *retval;
 
 	schedparam.priority = CONFIG_NUM_COOP_PRIORITIES - 1;
+	min_prio = sched_get_priority_min(schedpolicy);
+	max_prio = sched_get_priority_max(schedpolicy);
+
+	if (min_prio < 0 || max_prio < 0 || schedparam.priority < min_prio ||
+	    schedparam.priority > max_prio) {
+		TC_START("Scheduling priority outside valid priority range\n");
+		goto done;
```

Comment:
This doesn't actually test the scheduler behavior, which is kind of implied in the commit message.  If we are going to support POSIX scheduler behavior, we should be writing a test to make sure that it does what the API claims.

Reply:
yes i got it, I should have give this comment through TC_PRINT.